src/main/java/io/fabric8/maven/docker/config/ImageConfiguration.java

```diff
@@ -97,7 +97,7 @@ private void addVolumes(RunImageConfiguration runConfig, List<String> ret) {
 
     private void addLinks(RunImageConfiguration runConfig, List<String> ret) {
         // Custom networks can have circular links, no need to be considered for the starting order.
-        if (runConfig.getLinks() != null && !runConfig.getNetworkingConfig().isCustomNetwork()) {
```

Comment:
I've added @Nonnull annotations to show the deduction chain.  EnvUtil.splitAtCommasAndTrim(Iterable<String> input) never returns null.